### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by aljones15**
Adds eslint to the project.
Adds a check that a validator inside a module is a function.
The later helps out when dealing eslint files and stuff like that are in the same dir
as the validators.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -57,9 +57,11 @@ function init() {
           if(tmp in schemas) {
             logger.debug('overwriting schema "' + tmp + '" with ' + file);
           }
-          schemas[tmp] = module[key];
-          schemas[tmp].instance = schemas[tmp]();
-          logger.debug('loaded schema: ' + tmp);
+          if(typeof module[key] === 'function') {
```
Comment (mattcollier) [inline lib/index.js]: what happens if schema is not a function?

Comment (aljones15) [reply lib/index.js]: look at the next line, the issue is if we pass a schema that is not a function the current setup assumes it is a function.


https://github.com/digitalbazaar/bedrock-validation/blob/145c388b372832b39abdc408891f45a87e868d4b/lib/index.js#L55-L63

code in master fails if schema is an object right now.
I also might add the current way this works is pretty bad. a naming convention like validator.schema.js would be better so we could at least have some control over which files in a dir are inspected for schemas.

Comment (mattcollier) [reply lib/index.js]: The question I meant to ask was:  What happens **after this patch** if schema is not a function?

Comment (aljones15) [reply lib/index.js]: it is ignored.

**Comment by msporny** [comment]
@aljones15 @mattcollier What should we be doing with this PR? Merge or not?

**Comment by mattcollier** [comment]
Linting has already landed on `main`, so at minimum this needs to be rebased on the current `main` and merge conflicts resolved.

Subsequent to that, we're back to figuring out what issue this is aimed to solve and what the best way to solve it is.

**Comment by msporny** [comment]
This hasn't moved forward in 2 years, it's not moving forward, it's not clear what problem it solves... closing. We can try again if whatever this PR is attempting to address is/continues to be a problem.

**Pull request closed without merging**